app/uk/gov/hmrc/tai/viewModels/IncomeSourceSummaryViewModel.scala

```diff
@@ -66,21 +66,28 @@ object IncomeSourceSummaryViewModel {
   }
 
   private def companyBenefitViewModels(empId: Int, benefits: Benefits): Seq[CompanyBenefitViewModel] = {
+
     val ccBenVMs = benefits.companyCarBenefits collect {
       case ccBen: CompanyCarBenefit if ccBen.employmentSeqNo == empId =>
         val changeUrl = controllers.routes.CompanyCarController.redirectCompanyCarSelection(empId).url
         CompanyBenefitViewModel(Messages("tai.taxFreeAmount.table.taxComponent.CarBenefit"), ccBen.grossAmount, changeUrl)
     }
+
+    val medBenVMs = benefits.otherBenefits collect {
+      case benefit: GenericBenefit if benefit.benefitType == MedicalInsurance && benefit.employmentId.isDefined && benefit.employmentId.get == empId =>
```

Comment:
Can you use something like this instead?
case benefit @GenericBenefit(MedicalInsurance, Some(id), _) if id == empId =>